Skip tags that are mapped to empty

renemarc opened the pull request:
Currently, all webhook requests from Docker Hub are being transformed into CI requests. However, it would be useful to drop some requests for tags that should not be processed, for instance when trying to prevent circular build loops between Travis CI and Docker Hub. One could use TAG_MAPPINGS to map them to non-existant git branches, but this still sends a CI request ...and Travis CI has a pretty low rate limit to start with.

This proposed solution allows for empty values in TAG_MAPPINGS that would then get skipped. Say, when using `TAG_MAPPINGS="prod=master;test=master;develop="` then only a webhook call containing `prod` or `test` would trigger a build request for the `master` branch while the `develop` webhook call would be skipped.

Do feel free to edit this PR for style and function, as I am less familiar with Ruby. 😃

renemarc committed:
Skip tags mapped to empty strings, optionally disallow passthrough

Includes related test cases

renemarc commented:
I've updated this PR to allow disabling the passthrough of Dockerhub tags. This way, only Dockerhub tags mapped with TAG_MAPPINGS will be used, and any non-mapped tags will not trigger a CI build. In my case, this allows preventing git pushes to feature branches from triggering unnecessary builds.

This is optional, and the default behavior is not altered if this environment variable isn't set and enabled.

dwilkie commented:
@renemarc thanks for the PR. Some of the automated checks are failing regarding test coverage. Can you have a go at writing a test case for this new feature?

See https://github.com/somleng/dockerhub2ci/blob/develop/spec/models/webhook_subscriber/travis_spec.rb for more info

renemarc commented:
@dwilkie I've added 3 scenarios, hopefully that covers all possibilities. :)

Since the tests have passed, I've squashed the commits on my branch for easier reviewing and merging.
